dfinity/nns-dapp
# Sync svelte login and logout

# Motivation
If a user logs in on one tab, they should be logged in on all tabs, likewise if a user logs out on one tab, they should be logged out on all.  There is a security ticket regarding logging out:  https://dfinity.atlassian.net/browse/FOLLOW-229

# Changes
- Create a broadcast mechanism - this is a channel shared between tabs on the same origin that can be used for them to communicate.
- If one tab logs out, it broadcasts the fact, others listen to the channel and log out as well.
- If one tab logs in, the other tabs initialize their login status; assuming the login is real they will also log in.

# Testing
- Start a server with `frontend/svelte$ nns run dev`
- Open three tabs
- Log in to any one.
- Verify that all three tabs log in
- Log out of any one
- Verify that all three tabs log out

Tested on Chrome and Firefox.  Simulated Safari and IE by setting the check for BroadcastChannel to false.
